core/src/test/scala/wartremover/warts/StringPlusAnyTest.scala

```diff
@@ -38,9 +38,16 @@ class StringPlusAnyTest extends FunSuite with ResultAssertions {
     assertEmpty(result)
   }
 
+  test("inserting into a Set is allowed") {
+    val result = WartTestTraverser(StringPlusAny) {
+      Set("") + ""
+    }
+    assertEmpty(result)
+  }
+
   test("custom-defined + is allowed") {
     val result = WartTestTraverser(StringPlusAny) {
-      class C { def +(s: String) = s }
```

Comment:
It looks like you broke this test and just changed the test rather than addressing the issue. I think this issue can be easily addressed by changing the cases we check for thusly:
```scala
   case Apply(Select(Apply(Select(Select(_, PredefName), Any2StringAddName), _), Plus), _) =>
   case t @ Apply(Select(lhs, Plus), List(rhs)) if isString(t) && isString(lhs) && !isStringExpression(rhs) && !isSynthetic(u)(t) =>
```

Anything of the form `(t : NotAString) + (s : String)` always passes through `Any2StringAdd` so we can easily handle that case with no false positives or negatives by just forbidding `Any2StringAdd` and focus our efforts on the reverse case of `(s : String) + (t : NotAString)`.